spec/consumer_spec.rb

```diff
@@ -224,6 +224,36 @@
       end
     end
 
+    context "when fetch_batches returns FetchError" do
+      let(:thread_timeout) { 3 }
+      before do
+        allow(cluster).to receive(:mark_as_stale!)
+        allow(consumer).to receive(:fetch_batches).and_raise(Kafka::FetchError)
+      end
+
+      it "runs forever if exit_if is not passed" do
+        th = Thread.new do
+          consumer.each_batch do |message|
+            fail "this block is never called"
+          end
+        end
+
+        expect(th.join(thread_timeout)).to be_nil # join returning nil is a sign of thread not completing within timeout
+        th.kill
+      end
+
+      it "aborts earlier if exit_if is passed" do
+        th = Thread.new do
+          started_at = Time.now
+          consumer.each_batch(exit_if: -> { Time.now > (started_at + 2) } ) do |message|
```

Comment:
I think you would get the same test coverage with `exit_if: ->{ true }`, but without making the tests 2 seconds slower